### DISCUSSION TITLE
a very terribale bug

### DISCUSSION THREAD

**Discussion by MrYZhou** [fastapi/fastapi#6277] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@router.get("/list",status_code=200)
async def index():
    with Session(engine) as session:
        list = session.exec(select(Config).offset(0).limit(100)).all()
        return list
```

### Description

it can not response a result, a very strange phenomenon. At the command line, the request url has been regnized by fastapi,but can not into breakline. without any error info.

finaly,i change the request method with post,it work. it is so unreasonable.
It is worth mentioning that the way can work normally in other router file.

### Operating System

Windows

### FastAPI Version

latest

### Python Version

3.10.5

### Additional Context

Consider that bugs are hard to reproduce,this is my demo url.

about demo

path: router/generate.py

> @router.get("/list",status_code=200) async def index(): with Session(engine) as session: list = session.exec(select(Config).offset(0).limit(100)).all() return list

[https://github.com/MrYZhou/fastapi-le.git](url)

**Comment by JarroVGIT** [comment]
Please try to unindent your return statement, it is now part of your context manager.

```python
@router.get("/list",status_code=200)
async def index():
    with Session(engine) as session:
        list = session.exec(select(Config).offset(0).limit(100)).all()
    return list
```
This way, the `__exit__` is called of `Session()` before the response is returned.

(please try to provide much clearer issue titles, this is very unreasonably vague)

**Comment by MrYZhou** [comment]
the reason is that i define a router rule like 

@router.get("/{id}", tags=["web", "app"])
async def index(id):
    pass

so ,the pattern cath the router rule
@router.get("/list",status_code=200)
async def index():
     with Session(engine) as session:
         list = session.exec(select(Config).offset(0).limit(100)).all()
         return list
